### DISCUSSION TITLE
How to work with embedded smart objects?

### DISCUSSION THREAD

**Discussion by PcAlien** [loonghao/photoshop-python-api#337] [Q&A]
Hi there,
I got a PSD file with an embedded (not linked) smart object layer.
I need to modify that smart object, however, I never found out how to open it as a document. The only thing I can do is to handle it like every other normal layer, but that won't help.
Any ideas?

Thanks a lot!

**Comment by wiloaguilar** [comment]
Hi, you need to open the temp file or psb file, you can check it out when you open the smart object layer in photoshop. this is an example path: psd_file = "C:/Users/User/AppData/Local/Temp/Layer 11.psb". But you can not save it this because the save psb fuction is not available
